frappe/email/doctype/newsletter/newsletter.py

```diff
@@ -132,33 +132,15 @@ def get_email_groups(name):
 
 
 @frappe.whitelist(allow_guest=True)
-def unsubscribe(email, name):
-	if not verify_request():
-		return
-
-	primary_action = frappe.utils.get_url() + "/api/method/frappe.email.doctype.newsletter.newsletter.confirmed_unsubscribe"+\
-		"?" + get_signed_params({"email": email, "name":name})
-	return_confirmation_page(email, name, primary_action)
-
-
-@frappe.whitelist(allow_guest=True)
-def confirmed_unsubscribe(email, name):
-	if not verify_request():
-		return
-
-	for email_group in get_email_groups(name):
-		frappe.db.sql('''update `tabEmail Group Member` set unsubscribed=1 where email=%s and email_group=%s''',(email, email_group.email_group))
-
-	frappe.db.commit()
-	return_unsubscribed_page(email, name)
-
-def return_confirmation_page(email, name, primary_action):
-	frappe.respond_as_web_page(_("Unsubscribe from Newsletter"),_("Do you want to unsubscribe from this mailing list?"),
-		indicator_color="blue", primary_label = _("Unsubscribe"), primary_action=primary_action)
-
-def return_unsubscribed_page(email, name):
-	frappe.respond_as_web_page(_("Unsubscribed from Newsletter"),
-		_("<b>{0}</b> has been successfully unsubscribed from this mailing list.").format(email, name), indicator_color='green')
+def confirmed_unsubscribe(email, group):
+	""" unsubscribe the email(user) from the mailing list(email_group) """
+	frappe.flags.ignore_permissions=True
+	doc = frappe.get_all('Email Group Member', filters={"email": email, "email_group": group}, fields=['name','unsubscribed'], ignore_permissions = True)
```

Comment:
ok, removed the ignore_permission

Reply:
Actually, I think `get_doc` need two parameters `doctype` and `name ` to load specific doc (As given in documentation) .
So ,used `get_all` to fetch the name and used it to fetch and update doctype docs.

Reply:
`get_doc` also takes `doctype` and `filters` as its arguments.

For example:

```python
user = frappe.get_doc("User", {"first_name": "Test"})
```

will return the User object if it exists in the system, otherwise throw a `DoesNotExistError`.

Reply:
thanks, done